Task3/src/Services/GenericService.hs

```diff
@@ -0,0 +1,50 @@
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Services.GenericService (GenericService(..)) where
+
+import Repositories.GenericRepository.GenericRepository as R
+import Mappings.Mappings (Mapping(..))
+import Mappings.MappingParams (MappingParams(..))
+import Services.SearchService (SearchService(..))
+import Data.ServiceEntity ( ServiceEntity )
+import Data.Entities (Customer, Order, Product, Shop)
+import Data.Models (CustomerModel, OrderModel, ProductModel, ShopModel)
+import Utils.Utils (unwrap)
+
+class  (GenericRepository a, ServiceEntity b) => GenericService a b where
```

Comment:
Let's remove 'b' from the definition of the service. 

So even if you define:

`implement GenericService Customer`

it still should work as

```
S.getList @Customer @CustomerDetailedModel
S.getList @Customer @CustomerSimpleModel
```